products/memgraph/v0.15.0/reference_guide/other-features.md

```diff
@@ -159,3 +159,150 @@ to true is matched:
+operator(s) to the top-most logical operator(s). The execution of the query
+proceeds iteratively, from the bottom to the top.
+
+In the example above, the resulting plan is a pipeline of 3 logical operators.
+`Once` is the identity logical operator which does nothing and is always found
+at the "start" of the pipeline; `ScanAll` is a logical operator which produces
+all of the nodes in the graph, one by one; and `Produce` is a logical operator
+which takes data produced by another logical operator and produces data for the
+query's result set.
+
+A slightly more complicated example would be:
+
+```opencypher
+EXPLAIN MATCH (n :Node)-[:Edge]-(m :Node) WHERE n.prop = 42 RETURN *
+```
+
+```
++--------------------------------+
+| QUERY PLAN                     |
++--------------------------------+
+|  * Produce {m, n}              |
+|  * Filter                      |
+|  * Expand (m)-[anon1:Edge]-(n) |
+|  * ScanAllByLabel (n :Node)    |
+|  * ScanAllByLabel (m :Node)    |
+|  * Once                        |
++--------------------------------+
+```
+
+In this example, the `Filter` logical operator is used to filter the matched
+nodes because of the `WHERE n.prop = 42` construct. The `Expand` logical
+operator is used to find an edge between two nodes, in this case `m` and `n`
+which were matched previously using the `ScanAllByLabel` logical operator (a
+variant of the `ScanAll` logical operator mentioned previously).
+
+The execution of the query proceeds iteratively as follows. First, two vertices
+of type `:Node` are found as the result of the two scans. Then, we try to find a
+path that consists of the two vertices and an edge between them. If a path is
+found, it is further filtered based on a property of one of the vertices.
+Finally, if the path satisfied the filter, its two vertices are added to the
+query's result set.
+
+A simple example showcasing the fully general tree structure of the plan could
+be:
+
+```opencypher
+EXPLAIN MERGE (n) RETURN n
+```
+
+```
++------------------+
+| QUERY PLAN       |
++------------------+
+|  * Produce {n}   |
+|  * Accumulate    |
+|  * Merge         |
+|  |\ On Match     |
+|  | * ScanAll (n) |
+|  | * Once        |
+|  |\ On Create    |
+|  | * CreateNode  |
+|  | * Once        |
+|  * Once          |
++------------------+
+```
+
+The `Merge` logical operator (constructed as a result of the `MERGE` construct)
+can take input from upto 3 places. The `On Match` and `On Create` branches are
+"pulled from" only if a match was found or if a new vertex has to be created,
+respectively.
+
+### Profiling queries
+
+Along with inspecting a query's plan as described in the section [Profiling
+queries](#profiling-queries), it is also possible to profile the execution of a
+query and get a detailed report on how the query's plan behaved. For every
+logical operator the following info is provided:
+
+- `ACTUAL HITS` -- the number of times a particular logical operator was pulled
```

Comment:
-- is not going to be rendered properly in markdown. I guess you wanted to use an em-dash, you can do that with &mdash;

Reply:
I would also add dots at the end of the sentences but that's debatable :)

Reply:
> -- is not going to be rendered properly in markdown. I guess you wanted to use an em-dash, you can do that with —

I usually use en-dashes in lists like these. I'm not sure whether that's a valid thing to do according to standard typographical conventions or something, but if it's not going to be rendered properly anyway then I'll use an em-dash.